common/db.go

```diff
@@ -19,15 +23,18 @@ type DB struct {
 	dbh *sql.DB
 
 	// prepared statements
-	selectOrchs                *sql.Stmt
-	filterOrchs                *sql.Stmt
-	updateOrch                 *sql.Stmt
-	updateKV                   *sql.Stmt
-	insertUnbondingLock        *sql.Stmt
-	useUnbondingLock           *sql.Stmt
-	unbondingLocks             *sql.Stmt
-	withdrawableUnbondingLocks *sql.Stmt
-	insertWinningTicket        *sql.Stmt
+	selectOrchs                      *sql.Stmt
+	updateOrch                       *sql.Stmt
+	updateKV                         *sql.Stmt
+	insertUnbondingLock              *sql.Stmt
+	useUnbondingLock                 *sql.Stmt
+	unbondingLocks                   *sql.Stmt
+	withdrawableUnbondingLocks       *sql.Stmt
+	insertWinningTicket              *sql.Stmt
+	insertMiniHeader                 *sql.Stmt
+	findLatestMiniHeader             *sql.Stmt
+	findAllMiniHeadersSortedByNumber *sql.Stmt
```

Comment:
I thought it would be nice for the queries to be named according to the exported method that implements those queries. While we could rename the queries, we can't rename the exported methods as they need to adhere to the `MiniHeaderStore` interface introduced in https://github.com/livepeer/go-livepeer/pull/1030/files

Re-orgs are handled by `blockwatch.buildCanonicalChain()` which reorganizes the stack kept in the DB through the methods that are added here. 

> Just curious how reorgs are planned to be handled here?

Re-orgs are handles by the blockwatch package itself (`blockwatch.buildCanonicalChain()`), which maintains a stack of blocks. in this PR we implement only a persistant storage for that stack. 
If we find a re-org we will pop off blocks off the stack until the point of the re-org and then start Pushing blocks again. 

eg. `Stack.Pop` uses `DB..FindLatestMiniHeader` and then `DB.DeleteMiniHeader(latestMiniHeader.Hash)` but DB doesn't need to know about any of the reorg logic.

Reply:
Sounds like the interface could also be renamed if we're just importing everything :) But that's me - just renaming the query handle itself is fine here too.

Thanks for the explanation on reorgs! I would have thought it would build all chains as a tree and use the longest / deepest as the canonical chain, but I guess it makes sense to prune the DB occasionally if a reorg is detected.

Reply:
Yeah we could have, but we decided to move fast and just port blockwatch package over from 0x-mesh and remove dependencies on their codebase. The latter included moving away from a specific database implementation they used to an interface that defines the same methods so we can plug in our own DB. 

0x will make their blockwatch package standalone in the coming weeks which we'll then contribute back to, this is more of an intermediate solution that will behave very similar to the eventual API.  I also had a few optimisations in mind but in today's streamflow design meeting we decided to hold off on those to not diverge from the original API too much and focus in implementing the livepeer specific event watching logic first.